lib/discourse_reactions/topic_view_serializer_extension.rb

```diff
@@ -3,7 +3,10 @@
 module DiscourseReactions::TopicViewSerializerExtension
   def posts
     if SiteSetting.discourse_reactions_enabled
-      posts = object.posts.includes(:post_actions, reactions: { reaction_users: :user })
+      posts = object.posts
+        .includes(:post_actions, reactions: { reaction_users: :user })
+        .to_a
```

Comment:
We need to load this upfront and set `posts` to the loaded records. Otherwise, we end up working with an ActiveRecord association which will trigger a DB query each time we iterate through it.